### PR TITLE
always enable kms encryption in fscloud

### PR DESCRIPTION
**Pull request by Aayush-Abhyarthi**
### Description

https://github.com/terraform-ibm-modules/terraform-ibm-cos/issues/487

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by ocofaigh — COMMENTED** [review]
The `bucket_configs` variable still has an option called `kms_encryption_enabled`. However if we remove it, it means the format of the variable is now out of sync with the buckets module, so I think this might be why it was left in with a validation block. I'm thinking we don't make change here at all, so the format of the variable is the same. 
@daniel-butler-irl I think you implemented this - would you agree?

**Comment by daniel-butler-irl** [comment]
Yes, this was to keep the same bucket_configs configuration across all implementations. I do not have any strong opinions on keeping them the same or making them different. If you want to make a decision or bring it up on a deep-dive?

**Comment by ocofaigh** [comment]
Can we add variable validation (at the variable level) to ensure that value of `kms_encryption_enabled` is true. It would be better than catching it later like current logic does. Want to try that @Aayush-Abhyarthi ?

File: modules/fscloud/variables.tf [hunk 1]
```diff
@@ -131,6 +131,11 @@ variable "bucket_configs" {
   }))
   description = "Cloud Object Storage bucket configurations"
   default     = []
+
+  validation {
+    condition     = length([for bucket_config in var.bucket_configs : true if contains([true], bucket_config.kms_encryption_enabled)]) == length(var.bucket_configs)
+    error_message = "fscloud mandates kms encryption."
```
Comment (ocofaigh) [inline modules/fscloud/variables.tf:137]: The FSCloud submodule mandates that `kms_encryption_enabled` is set to `true` for all buckets in `bucket_configs` input variable value

**Review by ocofaigh — CHANGES_REQUESTED** [review]
left a comment. Also we can remove this now right? https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/04cbc060e805e98b5d315f605c3d9e774c247f50/modules/fscloud/main.tf#L9

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**